domain/schema/model/sql/0021-network.sql

```diff
@@ -36,23 +36,18 @@ CREATE TABLE link_layer_device (
     net_node_uuid TEXT NOT NULL,
     name TEXT NOT NULL,
     mtu INT,
-    -- will be NULL for a placeholder link layer device.
+    -- NULL for a placeholder link layer device.
     mac_address TEXT,
-    -- provider_id comes from the provider, no FK.
-    provider_id TEXT,
-    -- one of loopback, ethernet, bridge etc.
     device_type_id INT NOT NULL,
-    -- one of nonvirtualport, openvswitch
     virtual_port_type_id INT NOT NULL,
-    -- true if the device should be activated on boot.
-    is_auto_start BOOLEAN NOT NULL DEFAULT false,
-    -- true when the device is up (enabled).
+    -- True if the device should be activated on boot.
+    is_auto_start BOOLEAN NOT NULL DEFAULT true,
+    -- True when the device is in the up state.
     is_enabled BOOLEAN NOT NULL DEFAULT true,
-    -- the link layer device which is the parent to this device.
-    parent_device_uuid TEXT,
-    CONSTRAINT fk_link_layer_device_parent_device
-    FOREIGN KEY (parent_device_uuid)
-    REFERENCES link_layer_device (uuid),
+    -- True if traffic is routed out of this device by default.
+    is_default_gateway BOOLEAN NOT NULL DEFAULT false,
+    -- IP address of the default gateway.
+    gateway_address TEXT,
```

Comment:
Why not have this as a separate table? We can avoid the boolean this way.

Reply:
It could even be a join table with uuids from lld and ip_address.

Reply:
And the contents should be sparse with respect to llds or ip addresses.

Reply:
You can have a gateway address on a NIC without the NIC being the default NIC. And that address is a next hop - we won't necessarily have it in the `ip_address` table.